testutil/file.go

```diff
@@ -0,0 +1,117 @@
+package testutil
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+// FileInfo is a handy struct for asserting information related to file.
+// TODO: use continuity? ( requires https://github.com/stevvooe/continuity/issues/48 )
+type FileInfo struct {
```

Comment:
If we change the setup to use the apply methods we can use the method of applying inside the snapshot, then applying a temp directory and comparing the result. I prefer that method to defining the file info like this, makes it much easier to generate more complicated tests without exploding the amount of test definition.